source/common/protobuf/utility.cc

```diff
@@ -424,6 +497,22 @@ std::string MessageUtil::CodeEnumToString(ProtobufUtil::error::Code code) {
   }
 }
 
+ProtobufWkt::Value ValueUtil::loadFromYaml(const std::string& yaml) {
+  try {
+    return parseYamlNode(YAML::Load(yaml));
+  } catch (YAML::ParserException& e) {
+    throw EnvoyException(e.what());
+  } catch (YAML::BadConversion& e) {
+    throw EnvoyException(e.what());
+  } catch (std::exception& e) {
+    // There is a potentially wide space of exceptions thrown by the YAML parser,
+    // and enumerating them all may be difficult. Envoy doesn't work well with
+    // unhandled exceptions, so we capture them and record the exception name in
+    // the Envoy Exception text.
+    throw EnvoyException(fmt::format("Unexpected YAML exception: {}", +e.what()));
```

Comment:
This seems like a candidate for refactoring with `MessageUtil::loadFromYaml`.

Reply:
I tried and reverted, because parseYamlNode can raise YAML::Exception too.